packages/shared_preferences/shared_preferences_android/android/src/main/kotlin/io/flutter/plugins/sharedpreferences/SharedPreferencesPlugin.kt

```diff
@@ -31,6 +31,7 @@ import kotlinx.coroutines.runBlocking
 const val TAG = "SharedPreferencesPlugin"
 const val SHARED_PREFERENCES_NAME = "FlutterSharedPreferences"
 const val LIST_PREFIX = "VGhpcyBpcyB0aGUgcHJlZml4IGZvciBhIGxpc3Qu"
+const val JSON_LIST_PREFIX = "VGhpcyBpcyB0aGUgcHJlZml4IGZvciBhIGxpc3Qu!"
```

Comment:
Does this need to be the same value as LegacySharedPreferencesPlugin? If so can you add  comment as to why and that they need to be identical? 

Same comments as above about using similar strings. I think this should be a different value entirely.

Reply:
The legacy system and the new system can both read from the same files, so we need the same prefixes